### DISCUSSION TITLE
Generate migrations from blueprint changes

### DISCUSSION THREAD

**Discussion by duncanmcclean** [statamic-rad-pack/runway#519] [Feature Requests]
A feature I've been thinking about recently is automatically generating migrations whenever fields are added/updated/removed from the Runway blueprint.

While this feature would be amazing and seems easy at the surface, there's quite a few gotchas that make this idea quite complicated:

* Fieldsets - we'd need to watch for changes in any fieldsets used in your Runway blueprint (& recursively too).
* When you change "Max Items" on a field, which causes the column type to change, what happens to existing data?
    * The "String -> JSON" conversion is easy.
    * However, the "JSON -> String" conversion isn't... what would happen to existing data w/ more than a single item in the JSON array?
* How can we track a field being renamed? 
    * It'd be picked up as a field being removed and a new field being added. However... then existing data for that field would be deleted.

**Comment by nopticon** [comment]
This feature would be great to manage table fields easily.
